src/core/src/runtime/string_aligned_buffer.cpp

```diff
@@ -108,6 +111,12 @@ SharedStringAlignedBuffer::SharedStringAlignedBuffer(char* ptr, size_t size) {
     m_num_elements = size / ov::element::string.size();
 }
 
+SharedStringAlignedBuffer::~SharedStringAlignedBuffer() {
+    // to prevent deallocation in parent dtors.
+    m_aligned_buffer = nullptr;
+    m_allocated_buffer = nullptr;
```

Comment:
why is it better? why not to zero other members?

Reply:
There is no need to zero of all members as object will not exist any more (the dtor is simpler).
Just pointers have to be null to prevent resources release when parent dtor called.